packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/FlexibleSyncIntegrationTests.kt

```diff
@@ -289,4 +296,45 @@ class FlexibleSyncIntegrationTests {
             assertEquals("blueEmbedded", obj.embedded!!.embeddedName)
         }
     }
+
+    @Test
+    fun compensationWrite_writeOutsideOfSubscriptionsGetsReveredByServer() {
+        val user1 = app.createUserAndLogin()
+
+        val channel = Channel<CompensatingWriteException>(1)
+
+        val config1 = SyncConfiguration.Builder(user1, defaultSchema)
+            .errorHandler { _: SyncSession, syncException: SyncException ->
+                runBlocking {
+                    channel.send(syncException as CompensatingWriteException)
+                }
+            }
+            .build()
+
+        runBlocking {
+            val expectedPrimaryKey = BsonObjectId()
+
+            Realm.open(config1).use { realm ->
+                realm.subscriptions.update {
+                    add(realm.query<FlexParentObject>("_id = $0", BsonObjectId()))
```

Comment:
I would move this below `expectedPrimaryKey` and assert that the two IDs are different

Reply:
We cannot move it there because you need at least a subscription to be able to write:

`[RLM_ERR_NO_SUBSCRIPTION_FOR_WRITE]: Cannot write to class FlexParentObject when no flexible sync subscription has been created.`

Reply:
Sorry, I meant just the `BsonObjectID()` you create here. The test assumes it is different than the first one. It would be nice if that assumption was explicit.

Reply:
I think it is not required. Random `ObjectId` creation has a counter (3 bytes) parameter that is incremented on each new `ObjectId`, so there should not be clashes between two consecutive random `ObjectId`.

Reply:
Ah yeah, good point